alerting/multiorg_alertmanager.go

```diff
@@ -2,332 +2,11 @@ package alerting
-	dataDir := filepath.Join(moa.settings.DataPath, workingDir)
-	files, err := ioutil.ReadDir(dataDir)
-	if err != nil {
-		moa.logger.Error("failed to list local working directory", "dir", dataDir, "err", err)
-		return
-	}
-	for _, file := range files {
-		if !file.IsDir() {
-			moa.logger.Warn("ignoring unexpected file while scanning local working directory", "filename", filepath.Join(dataDir, file.Name()))
-			continue
-		}
-		orgID, err := strconv.ParseInt(file.Name(), 10, 64)
-		if err != nil {
-			moa.logger.Error("unable to parse orgID from directory name", "name", file.Name(), "err", err)
-			continue
-		}
-		_, exists := activeOrganizations[orgID]
-		if !exists {
-			moa.logger.Info("found orphan organization directory", "orgID", orgID)
-			workingDirPath := filepath.Join(dataDir, strconv.FormatInt(orgID, 10))
-			fileStore := NewFileStore(orgID, moa.kvStore, workingDirPath)
-			// Cleanup all the remaining resources from this alertmanager.
-			fileStore.CleanUp()
-		}
-	}
-	// Remove all orphaned items from kvstore by listing all existing items
-	// in our used namespace and comparing them to the currently active
-	// organizations.
-	storedFiles := []string{notificationLogFilename, silencesFilename}
-	for _, fileName := range storedFiles {
-		keys, err := moa.kvStore.Keys(ctx, kvstore.AllOrganizations, KVNamespace, fileName)
-		if err != nil {
-			moa.logger.Error("failed to fetch items from kvstore", "err", err,
-				"namespace", KVNamespace, "key", fileName)
-		}
-		for _, key := range keys {
-			if _, exists := activeOrganizations[key.OrgId]; exists {
-				continue
-			}
-			err = moa.kvStore.Del(ctx, key.OrgId, key.Namespace, key.Key)
-			if err != nil {
-				moa.logger.Error("failed to delete item from kvstore", "err", err,
-					"orgID", key.OrgId, "namespace", KVNamespace, "key", key.Key)
-			}
-		}
-	}
-}
-
-func (moa *MultiOrgAlertmanager) StopAndWait() {
-	moa.alertmanagersMtx.Lock()
-	defer moa.alertmanagersMtx.Unlock()
-
-	for _, am := range moa.alertmanagers {
-		am.StopAndWait()
-	}
-
-	p, ok := moa.peer.(*cluster.Peer)
-	if ok {
-		moa.settleCancel()
-		if err := p.Leave(10 * time.Second); err != nil {
-			moa.logger.Warn("unable to leave the gossip mesh", "err", err)
-		}
-	}
-}
-
-// AlertmanagerFor returns the Alertmanager instance for the organization provided.
-// When the organization does not have an active Alertmanager, it returns a ErrNoAlertmanagerForOrg.
-// When the Alertmanager of the organization is not ready, it returns a ErrAlertmanagerNotReady.
-func (moa *MultiOrgAlertmanager) AlertmanagerFor(orgID int64) (*Alertmanager, error) {
-	moa.alertmanagersMtx.RLock()
-	defer moa.alertmanagersMtx.RUnlock()
-
-	orgAM, existing := moa.alertmanagers[orgID]
-	if !existing {
-		return nil, ErrNoAlertmanagerForOrg
-	}
-
-	if !orgAM.Ready() {
-		return orgAM, ErrAlertmanagerNotReady
-	}
-
-	return orgAM, nil
-}
-
 // NilPeer and NilChannel implements the Alertmanager clustering interface.
 type NilPeer struct{}
```

Comment:
This type is def useful on this layer - `NilPeer` represents a [null object](https://en.wikipedia.org/wiki/Null_object_pattern) for the clustering interface of the Alertmanager. However, I think you're right that this might not be its home - I've made a note in #8